cache_flat_mutation_reader.hh

```diff
@@ -851,8 +851,9 @@ inline
 void cache_flat_mutation_reader::do_add_to_buffer(range_tombstone_change&& rtc) {
     clogger.trace("csm {}: push({})", fmt::ptr(this), rtc);
     position_in_partition::less_compare less(*_schema);
+    auto lower_bound_changed = less(_lower_bound, rtc.position());
     _lower_bound = position_in_partition(rtc.position());
-    _lower_bound_changed = less(_lower_bound, rtc.position());
+    _lower_bound_changed = lower_bound_changed;
     push_mutation_fragment(*_schema, _permit, std::move(rtc));
```

Comment:
btw, I think std::bad_alloc from push_mutation_fragment() will make the reader skip some range because _lower_bound was updated and the thing runs inside allocating section, which retries.